solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetContentStatusFilter.java

```diff
@@ -17,79 +17,61 @@
-        Map<String, List<String>> triples = dip.getTriples();
-
-        List<String> contentStatus = new ArrayList<String>();
-        setContentStatus(dip, triples, contentStatus);
-
+        List<String> contentStatus = determineContentStatus(dip);
         dip.getDocument().setContentStatus(contentStatus);
 
-        log.debug("Content status for {} set to {}", dip.getPid().getPid(), contentStatus);
+        log.debug("Content status for {} set to {}", dip.getPid().toString(), contentStatus);
     }
 
-    private void setContentStatus(DocumentIndexingPackage dip, Map<String, List<String>> triples, List<String> status)
+    private List<String> determineContentStatus(DocumentIndexingPackage dip)
             throws IndexingException {
-        List<String> datastreams = triples.get(FedoraProperty.disseminates.toString());
-        if (datastreams != null) {
-            String mdDescriptive = dip.getPid().getURI() + "/" + Datastream.MD_DESCRIPTIVE.getName();
-            boolean described = datastreams.contains(mdDescriptive);
-            if (described) {
-                status.add(FacetConstants.CONTENT_DESCRIBED);
-            } else {
-                status.add(FacetConstants.CONTENT_NOT_DESCRIBED);
-            }
-        }
 
-        // Valid/Not Valid content according to FITS
+        List<String> status = new ArrayList<>();
+        ContentObject obj = dip.getContentObject();
+        Resource resc = obj.getResource();
+
+        if (resc.hasProperty(Cdr.hasMods)) {
+            status.add(FacetConstants.CONTENT_DESCRIBED);
+        } else {
+            status.add(FacetConstants.CONTENT_NOT_DESCRIBED);
+        }
 
-        // Vocabulary validation
-        if (triples.containsKey(CDRProperty.invalidTerm.toString())) {
+        if (resc.hasProperty(Cdr.invalidTerm)) {
             status.add(FacetConstants.INVALID_VOCAB_TERM);
         }
 
-        // If its an aggregate, indicate if it has a default web object
-        List<String> contentModels = triples.get(FedoraProperty.hasModel.toString());
-        if (contentModels != null && contentModels.contains(Model.AGGREGATE_WORK.toString())) {
-            if (triples.containsKey(CDRProperty.defaultWebObject.toString())) {
-                status.add(FacetConstants.CONTENT_DEFAULT_OBJECT);
-            } else {
-                status.add(FacetConstants.CONTENT_NO_DEFAULT_OBJECT);
-            }
-        } else {
-            // Check to see if this object is the DefaultWebObject for its parent
-            DocumentIndexingPackage parentDip = dip.getParentDocument();
-            String parentsDWO = parentDip.getFirstTriple(CDRProperty.defaultWebObject.toString());
-            if (parentsDWO != null) {
-                if (dip.getPid().equals(new PID(parentsDWO))) {
-                    status.add(FacetConstants.CONTENT_IS_DEFAULT_OBJECT);
-                }
+        if (obj instanceof WorkObject) {
+            if (((WorkObject) obj).getPrimaryObject() == null) {
```

Comment:
The `instanceof` bit would still communicate that it only applies to works, and could be kept either way.  Its mainly a readability/maintainability decision versus minor performance hit (once caching is in place).  Another approach would be to add a "hasPrimaryObject" method to the WorkObject class.  But I'm okay with keeping things the way they are